client/src/components/Form/Elements/FormCheck.vue

```diff
@@ -25,26 +23,15 @@ const currentValue = computed({
         } else {
             emit("input", null);
         }
-        if (newValue.length === 0) {
-            selectAll.value = false;
-            indeterminate.value = false;
-        } else if (newValue.length === props.options.length) {
-            selectAll.value = true;
-            indeterminate.value = false;
-        } else {
-            indeterminate.value = true;
-        }
     },
 });
 
-const hasOptions = computed(() => {
-    return props.options.length > 0;
-});
-
-const selectAll = ref(false);
+const hasOptions = computed(() => props.options.length > 0);
+const indeterminate = computed(() => ![0, props.options.length].includes(currentValue.value.length));
+const selectAll = computed(() => currentValue.value.length === props.options.length);
 
-function onSelectAll() {
-    if (selectAll.value) {
+function onSelectAll(selected: boolean): void {
```

Comment:
The void is still unnecessary, but not worth burning CI time over.

Reply:
Interesting topic, maybe worth briefly talking about it during the next UX meeting to establish and communicate some best practices moving forward. Thank you.

Reply:
https://typescript-eslint.io/rules/no-inferrable-types/, we do extend `eslint:recommended`, so it should already be an error?!